From patches Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@redhat.com>
To: Anthony Liguori <anthony@codemonkey.ws>, Amit Shah <amit.shah@redhat.com>
Cc: Amit Shah <amit.shah@redhat.com>, qemu list <qemu-devel@nongnu.org>
Date: Mon,  1 Apr 2013 14:55:12 +0530
Message-id: 74d06e1f94cc2337aa1bb6784a7ad2210ec55364.1364808312.git.amit.shah@redhat.com
Subject: [PATCH 1/1] rng backend: open backend in blocking mode

Opening backends in non-blocking mode isn't necessary, we don't do
anything while waiting for data.

This also excuses us from checking for EAGAIN, which for the default
random backend, is a very common return error type.  Starting the guest
with '-device virtio-rng-pci', issuing a 'cat /dev/hwrng' in the guest
while also doing 'cat /dev/random' on the host causes

backends/rng-random.c:44:entropy_available: assertion failed: (len != -1)

without this fix.

Reported-by: yunpingzheng <yunzheng@redhat.com>
Signed-off-by: Amit Shah <amit.shah@redhat.com>
---
 backends/rng-random.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backends/rng-random.c b/backends/rng-random.c
index acd20af..252139b 100644
--- a/backends/rng-random.c
+++ b/backends/rng-random.c
@@ -74,7 +74,7 @@ static void rng_random_opened(RngBackend *b, Error **errp)
         error_set(errp, QERR_INVALID_PARAMETER_VALUE,
                   "filename", "a valid filename");
     } else {
-        s->fd = qemu_open(s->filename, O_RDONLY | O_NONBLOCK);
+        s->fd = qemu_open(s->filename, O_RDONLY);
 
         if (s->fd == -1) {
             error_set(errp, QERR_OPEN_FILE_FAILED, s->filename);
-- 
1.8.1.4

